File: 263-ugly-number/ken/263.py
class Solution(object):
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """

        if num == 0:
            return False

        primes = [2, 3 ,5]
        for prime in primes:
            while num % prime == 0:
                num /= prime

        return num == 1

File: 263-ugly-number/leochang/263.cpp
class Solution {
public:
    bool isUgly(int num) {
        if (0 >= num) return false;
        int primes[3] = {2, 3, 5};
        for (auto prime: primes) {
            while (0 == num % prime) {
                num = num / prime;
            }
        }
        return (1 == num);
    }
};
